hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/token/impl/CryptoPreTransactionHandlerImpl.java

```diff
@@ -52,19 +52,34 @@ public TransactionMetadata preHandleCryptoCreate(final TransactionBody tx) {
 
     @Override
     /** {@inheritDoc} */
-    public TransactionMetadata preHandleUpdateAccount(TransactionBody txn) {
-        throw new NotImplementedException();
+    public TransactionMetadata preHandleCryptoDelete(TransactionBody txn) {
+        final var op = txn.getCryptoDelete();
+        final var payer = txn.getTransactionID().getAccountID();
+        final var deleteAccountId = op.getDeleteAccountID();
+        final var transferAccountId = op.getTransferAccountID();
+        final var meta = new SigTransactionMetadata(accountStore, txn, payer);
+        if (meta.failed()) {
+            return meta;
+        }
```

Comment:
This seems suspicious. I assume this means that the constructor for SigTansactionMetadata is doing some kind of validation and setting the failed flag?

Reply:
I'm also worried about having to include this check in each pre-handle. If somebody forgets this check, is the pre-handle implementation wrong?

Reply:
Since every transaction's pre-transaction handler method have to lookup `payer` , payer is validated and added to `requiredKeys` in the constructor of `SigTransactionMetadata` . Do you suggest, having different method like this for clarity ?

```
final var meta = new SigTransactionMetadata(accountStore, txn);
addPayer(payer, meta);
```